src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1099,6 +1099,10 @@ UCS_CLASS_CLEANUP_FUNC(uct_dc_mlx5_ep_t)
     uct_dc_mlx5_ep_fc_cleanup(self);
     uct_dc_mlx5_ep_keepalive_cleanup(self);
 
+    if (self->dci != UCT_DC_MLX5_EP_NO_DCI) {
+        uct_dc_mlx5_iface_dci_free(iface, self);
+    }
+
```

Comment:
need to do it below the check for `uct_dc_mlx5_iface_is_dci_rand` below. With random policy single dci is shared among several eps

Reply:
also, seems we may not progress the arbiter of the pool of this released dci during the dci_release progress

Reply:
but we need to progress pending after releasing this dci

Reply:
ok, added